src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -246,34 +246,46 @@ enum class OpCase : uint8_t
 constexpr const auto & undec(const auto & x) { return x; }
 constexpr const auto & undec(const is_decimal auto & x) { return x.value; }
 
-template <typename A, typename B, typename Op, typename OpResultType = typename Op::ResultType>
+template <typename A, typename B, typename Op, typename OpResultType = typename Op::ResultType, bool is_compare=false>
```

Comment:
Do we need to encode the special case handling into the generic template interface? Such special case handling tends to accumulate over time and at some point the code becomes unmaintainable.

In places where we need to distinguish based on the operator type, we should instead check in-place, similar to this

```cpp
static constexpr bool is_compare = IsOperation<Op>::greatest || IsOperation<Op>::least;
```

(basically what you added elsewhere)